src/assets/src/components/queueSettings.tsx

```diff
@@ -0,0 +1,315 @@
+    const confirmRemoveHost = (h: User) => {
+        showConfirmation(dialogRef, () => doRemoveHost(h), "Remove Host?", `remove host ${h.username}`);
+    }
+    const addHost = async (uniqname: string) => {
+        const user = await confirmUserExists(uniqname);
+        recordQueueManagementEvent("Added Host");
+        await api.addHost(queue!.id, user.id);
+    }
+    const [doAddHost, addHostLoading, addHostError] = usePromise(addHost);
+
+    const removeQueue = async () => {
+        recordQueueManagementEvent("Removed Queue");
+        await api.deleteQueue(queue!.id);
+        location.href = '/manage';
+    }
+    const [doRemoveQueue, removeQueueLoading, removeQueueError] = usePromise(removeQueue);
+    const confirmRemoveQueue = () => {
+        showConfirmation(dialogRef, () => doRemoveQueue(), "Delete Queue?", "permanently delete this queue");
+    }
+
+    // On change handlers
+    const handleNameChange = (newName: string) => {
+        setName(newName);
+        validateAndSetStringResult(newName, queueNameSchema, setNameValidationResult, true);
+        if (showSuccessMessage) setShowSuccessMessage(false);
+    };
+    const handleDescriptionChange = (newDescription: string) => {
+        setDescription(newDescription);
+        validateAndSetStringResult(newDescription, queueDescriptSchema, setDescriptValidationResult, true);
+        if (showSuccessMessage) setShowSuccessMessage(false);
+    };
+    const handleAllowedChange = (newAllowedBackends: Set<string>) => {
+        setAllowedMeetingTypes(newAllowedBackends);
+        validateAndSetMeetingTypesResult(newAllowedBackends, setAllowedValidationResult, queue)
+        if (showSuccessMessage) setShowSuccessMessage(false);
+    };
+
+    // On click handlers
+    const handleSaveGeneralClick = () => {
+        const curNameValidationResult = !nameValidationResult
+            ? validateAndSetStringResult(name, queueNameSchema, setNameValidationResult, true)
+            : nameValidationResult;
+        const curDescriptValidationResult = !descriptValidationResult
+            ? validateAndSetStringResult(description, queueDescriptSchema, setDescriptValidationResult, true)
+            : descriptValidationResult;
+        const curAllowedValidationResult = !allowedValidationResult
+            ? validateAndSetMeetingTypesResult(allowedMeetingTypes, setAllowedValidationResult, queue)
+            : allowedValidationResult;
+
+        if (!curNameValidationResult!.isInvalid && !curDescriptValidationResult!.isInvalid && !curAllowedValidationResult!.isInvalid) {
+            const nameForUpdate = name.trim() !== queue?.name ? name : undefined;
+            const descriptForUpdate = description.trim() !== queue?.description ? description : undefined;
+            const allowedForUpdate = !compareStringArrays(Array.from(allowedMeetingTypes), queue!.allowed_backends)
+                ? allowedMeetingTypes : undefined;
```

Comment:
I re-wrote `compareStringArrays` to use `Set`s, and I do think it's better. It didn't really feel like I could (or necessarily should) get rid of the utility function all together though. I based this on the MDN `symmetricDifference` function, as indicated in the comment. It would be nice if JavaScript had the nice operators for unions and intersections like Python does.